packages/contracts-bedrock/test/L2/L2StandardBridge.t.sol

```diff
@@ -25,43 +25,91 @@ import { IL2StandardBridge } from "interfaces/L2/IL2StandardBridge.sol";
 contract L2StandardBridge_TestInit is CommonTest {
     /// @notice Sets up expected calls and emits for a successful ERC20 withdrawal.
-    function _preBridgeERC20(bool _isLegacy, address _l2Token) internal {
-        // Alice has 100 L2Token
-        deal(_l2Token, alice, 100, true);
-        assertEq(ERC20(_l2Token).balanceOf(alice), 100);
+    function _expectERC20WithdrawalCalls(
+        uint256 _amount,
+        uint32 _minGasLimit,
+        address _to,
+        bool _isLegacy,
+        address _l2Token
+    )
+        internal
+    {
+        deal(_l2Token, alice, _amount, true);
+        assertEq(ERC20(_l2Token).balanceOf(alice), _amount);
+
+        _expectWithdrawalCallsAndEvents(_amount, _minGasLimit, _to, _isLegacy, _l2Token);
+
+        vm.prank(alice, alice);
+    }
+
+    function _expectWithdrawalCallsAndEvents(
+        uint256 _amount,
+        uint32 _minGasLimit,
+        address _to,
+        bool _isLegacy,
+        address _l2Token
+    )
+        internal
+    {
         uint256 nonce = l2CrossDomainMessenger.messageNonce();
-        bytes memory message =
-            abi.encodeCall(IStandardBridge.finalizeBridgeERC20, (address(L1Token), _l2Token, alice, alice, 100, hex""));
-        uint64 baseGas = l2CrossDomainMessenger.baseGas(message, 1000);
+        bytes memory message = abi.encodeCall(
+            IStandardBridge.finalizeBridgeERC20, (address(L1Token), _l2Token, alice, _to, _amount, hex"")
+        );
+        uint64 baseGas = l2CrossDomainMessenger.baseGas(message, _minGasLimit);
         bytes memory withdrawalData = abi.encodeCall(
             ICrossDomainMessenger.relayMessage,
-            (nonce, address(l2StandardBridge), address(l1StandardBridge), 0, 1000, message)
-        );
-        bytes32 withdrawalHash = Hashing.hashWithdrawal(
-            Types.WithdrawalTransaction({
-                nonce: nonce,
-                sender: address(l2CrossDomainMessenger),
-                target: address(l1CrossDomainMessenger),
-                value: 0,
-                gasLimit: baseGas,
-                data: withdrawalData
-            })
+            (nonce, address(l2StandardBridge), address(l1StandardBridge), 0, _minGasLimit, message)
         );
 
+        _expectBridgeCalls(_amount, _minGasLimit, _isLegacy, _l2Token, message, baseGas, withdrawalData, _to);
+        _expectWithdrawalEvents(_amount, _minGasLimit, _to, _l2Token, message, nonce, baseGas, withdrawalData);
+    }
+
+    function _expectBridgeCalls(
+        uint256 _amount,
+        uint32 _minGasLimit,
+        bool _isLegacy,
+        address _l2Token,
+        bytes memory message,
+        uint64 baseGas,
+        bytes memory withdrawalData,
+        address _to
+    )
+        internal
+    {
         if (_isLegacy) {
-            vm.expectCall(
-                address(l2StandardBridge), abi.encodeCall(l2StandardBridge.withdraw, (_l2Token, 100, 1000, hex""))
-            );
+            if (_to == alice) {
```

Comment:
This is an unclear condition to discriminate between `withdraw` and `withdrawTo`. Maybe pass an enum as an argument to do that job.